### PR TITLE
Browser: refactor `mCards` partially

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
* split from https://github.com/ankidroid/Anki-Android/pull/14843/
  * This removes a number of references to `mCards`
  * We'll be changing this to type `CardOrNoteId` later, so combining reference methods is useful in reducing breakages  
* Reviewable, not mergeable as it's blocked on https://github.com/ankidroid/Anki-Android/pull/14900
  * After this goes in, some more `mCards` references may be removed

## How Has This Been Tested?
Unit test only

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt [hunk 1]
```diff
@@ -751,9 +751,13 @@ class CardBrowserTest : RobolectricTest() {
 
         browserWithNoNewCards.apply {
             searchAllDecks()
-            assertThat("Result should contain 4 cards", cardCount, equalTo(4))
+            assertThat("Result should contain 4 cards", viewModel.rowCount, equalTo(4))
             viewModel.setCardsOrNotes(NOTES)
-            assertThat("Result should contain 2 cards (one per note)", cardCount, equalTo(2))
+            assertThat(
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:756]: Fix

**Review by mikehardy — APPROVED** [review]
LGTM - needs a rebase to handle the related PR this was blocked on now that's merged
After rebase to main and clean CI squash merge +1

**Comment by david-allison** [comment]
@mikehardy rebased. Final 3 commits are new, as is an extraction from `RenderOnScroll`

**Review by mikehardy — APPROVED** [review]
still looks good 🚂

**Pull request merged**